rest/config.go

```diff
@@ -959,6 +984,24 @@ func ValidateConfigOrPanic(runMode SyncGatewayRunMode) {
 // It parses command-line flags, reads the optional configuration file, then starts the server.
 func ServerMain(runMode SyncGatewayRunMode) {
 	ParseCommandLine(runMode)
+
+	// Logging config will now have been loaded from command line
+	// or from a sync_gateway config file so we can validate the
+	// configuration and setup logging now
+	deferredLogs, err := config.setupAndValidateLogging()
+	if err != nil {
+		base.Fatalf(base.KeyAll, "Error setting up logging: %v", err)
+	}
+
+	// This is the earliest opportunity to print a startup indicator
+	// that will be persisted in log files
+	base.Broadcastf("==== %s ====", base.LongVersionString)
```

Comment:
Broadcast sounds like it's broadcasting over a network.  I think this would be clearer as `LogToAllTargets()` or `LogToAllDests()`

Reply:
It was difficult naming this when coming up with the logging API. In practice, this is the *only* place it is ever used, and should perhaps be removed in exchange for `base.PrintVersion()`.

The intention with `Broadcastf` was to cause second thought when writing it, as it circumvents a few user-defined log settings (i.e. `log_level=none` will still log these app startup indicators)

Reply:
Yeah I'd personally prefer it to be moved to `base.LogSyncGatewayVersion()` or `base.PrintSyncGatewayVersion()` over `base.Broadcastf()`.  "Broadcast" has different connotations .. like in networking or even condition variables for thread sync.  If we had to keep "Broadcast", maybe "LogBroadcast" would at least be slightly better.